agent/config/types.go

```diff
@@ -210,3 +210,24 @@ func (data *SensitiveRawMessage) UnmarshalJSON(jsonData []byte) error {
 	data.contents = json.RawMessage(jsonData)
 	return nil
 }
+
+// DefaultBool makes it possible to understand if a variable was set explicitly or relies on a default setting
+type DefaultBool struct {
+	value    bool
+	explicit bool
+}
+
+func NewDefaultBool(setting, explicit bool) DefaultBool {
```

Comment:
lint: this and other global methods need documentation

Reply:
looking over this again... should I have this be a private function and expose constants?

IE, ExplicitlyEnabled, ImplicitlyEnabled, ExplicitlyDisabled...